CenterForOpenScience/osf.io
# [IN-358][OSF] Collect existing registrations into OSF Registries

## Purpose

Collect existing public registrations into OSF Registries provider and add
logic to collect future registrations upon becoming public.

## Changes

- Add tests
- Add migration
- Add logic in sanctions to make sure registrations are collected after completing embargo both early termination and due date.

## QA Notes

- Immediately public registrations: should be collected right away, go to the registration after archiving
is over and you should see `OSF Registries's Collection` listed on overview page.
- Embargoed registrations: make sure they are not collected till the embargo completes and is terminated early.
- Retracted registrations: upon retraction/withdrawal the registration should be removed from OSF Registries.

## Documentation

## Side Effects

## Ticket

[[IN-358]](https://openscience.atlassian.net/browse/IN-358)
